mesonbuild/compilers/detect.py

```diff
@@ -1254,6 +1254,10 @@ def detect_nasm_compiler(env: 'Environment', for_machine: MachineChoice) -> Comp
             env.coredata.add_lang_args(comp_class.language, comp_class, for_machine, env)
             return comp_class([], comp, version, for_machine, info, cc.linker, is_cross=is_cross)
         elif 'Metrowerks' in output or 'Freescale' in output:
+            mwas_ver_match = re.search(r'Version (\d+)\.(\d+)\.?(\d+)? build (\d+)', output)
+            assert mwas_ver_match is not None, 'for mypy'  # because mypy *should* be complaning that this could be None
```

Comment:
Should this be a raised exception instead or do other code paths also assert the regexes work? `complaining` is misspelled